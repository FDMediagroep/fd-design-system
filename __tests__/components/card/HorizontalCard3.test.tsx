import React from 'react';
import { render } from '@testing-library/react';
import {
    HorizontalCard3,
    getCssClassNames,
} from '../../../src/components/card/HorizontalCard3';

describe('HorizontalCard3', () => {
    test('should render correctly', () => {
        const { container, getByText } = render(
            <HorizontalCard3
                id="1"
                intro="This is the intro of the card"
                label="Ondernemen"
                title="This is the title"
                time="11:50"
                url="https://example.com/1/This is the title"
            />
        );

        expect(getByText('This is the title')).toBeTruthy();
        expect(getByText('This is the intro of the card')).toBeTruthy();
        expect(getByText('Ondernemen')).toBeTruthy();
        expect(getByText('11:50')).toBeTruthy();
        expect(container.querySelector('[href]')?.getAttribute('href')).toBe(
            'https://example.com/1/This is the title'
        );
    });

    test('should render is-read correctly', () => {
        const { container } = render(
            <HorizontalCard3
                id="1"
                intro="This is the intro of the card"
                label="Ondernemen"
                title="This is the title"
                time="11:50"
                url="https://example.com/1/This is the title"
                isRead={true}
            />
        );
        expect(container).toMatchSnapshot();
    });

    test('should have correct class name', () => {
        const { container } = render(
            <HorizontalCard3
                id="1"
                intro="This is the intro of the card"
                label="Ondernemen"
                title="This is the title"
                time="11:50"
                url="https://example.com/1/This is the title"
            />
        );

        expect(
            container.firstElementChild?.classList.contains(
                getCssClassNames()[0]
            )
        ).toBeTruthy();
    });
});
