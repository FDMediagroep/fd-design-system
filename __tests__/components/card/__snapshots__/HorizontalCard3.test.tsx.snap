// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalCard3 should render is-read correctly 1`] = `
<div>
  <article
    class="undefined undefined xs__p+4"
    id="1"
  >
    <a
      aria-label="This is the title"
      href="https://example.com/1/This is the title"
    >
      <figure
        class="xs__m-0"
      >
        <picture>
          <img
            class="undefined "
            sizes="(min-width: 1080px) 244px, (min-width: 768px) 244px, 160px"
            src="https://images.example.com/image.png?auto=format&ixlib=react-9.10.0"
            srcset="https://images.example.com/image.png?auto=format&ixlib=react-9.10.0&w=100 100w,
https://images.example.com/image.png?auto=format&ixlib=react-9.10.0&w=116 116w,
https://images.example.com/image.png?auto=format&ixlib=react-9.10.0&w=135 135w,
https://images.example.com/image.png?auto=format&ixlib=react-9.10.0&w=156 156w,
https://images.example.com/image.png?auto=format&ixlib=react-9.10.0&w=181 181w,
https://images.example.com/image.png?auto=format&ixlib=react-9.10.0&w=210 210w,
https://images.example.com/image.png?auto=format&ixlib=react-9.10.0&w=244 244w,
https://images.example.com/image.png?auto=format&ixlib=react-9.10.0&w=283 283w,
https://images.example.com/image.png?auto=format&ixlib=react-9.10.0&w=328 328w,
https://images.example.com/image.png?auto=format&ixlib=react-9.10.0&w=380 380w,
https://images.example.com/image.png?auto=format&ixlib=react-9.10.0&w=441 441w,
https://images.example.com/image.png?auto=format&ixlib=react-9.10.0&w=512 512w,
https://images.example.com/image.png?auto=format&ixlib=react-9.10.0&w=594 594w,
https://images.example.com/image.png?auto=format&ixlib=react-9.10.0&w=689 689w,
https://images.example.com/image.png?auto=format&ixlib=react-9.10.0&w=799 799w,
https://images.example.com/image.png?auto=format&ixlib=react-9.10.0&w=927 927w,
https://images.example.com/image.png?auto=format&ixlib=react-9.10.0&w=1075 1075w,
https://images.example.com/image.png?auto=format&ixlib=react-9.10.0&w=1247 1247w,
https://images.example.com/image.png?auto=format&ixlib=react-9.10.0&w=1446 1446w,
https://images.example.com/image.png?auto=format&ixlib=react-9.10.0&w=1678 1678w,
https://images.example.com/image.png?auto=format&ixlib=react-9.10.0&w=1946 1946w,
https://images.example.com/image.png?auto=format&ixlib=react-9.10.0&w=2257 2257w,
https://images.example.com/image.png?auto=format&ixlib=react-9.10.0&w=2619 2619w,
https://images.example.com/image.png?auto=format&ixlib=react-9.10.0&w=3038 3038w,
https://images.example.com/image.png?auto=format&ixlib=react-9.10.0&w=3524 3524w,
https://images.example.com/image.png?auto=format&ixlib=react-9.10.0&w=4087 4087w,
https://images.example.com/image.png?auto=format&ixlib=react-9.10.0&w=4741 4741w,
https://images.example.com/image.png?auto=format&ixlib=react-9.10.0&w=5500 5500w,
https://images.example.com/image.png?auto=format&ixlib=react-9.10.0&w=6380 6380w,
https://images.example.com/image.png?auto=format&ixlib=react-9.10.0&w=7401 7401w,
https://images.example.com/image.png?auto=format&ixlib=react-9.10.0&w=8192 8192w"
          />
        </picture>
        <figcaption />
      </figure>
      <div
        class="undefined xs__pl+4 s__p+4"
      >
        <h1
          class="heading serif xs xs__m-0"
        >
          This is the title
        </h1>
      </div>
    </a>
  </article>
</div>
`;
